docs/en/ingest-management/security/certificates.asciidoc

```diff
@@ -0,0 +1,216 @@
+[[secure-connections]]
+[role="xpack"]
+= Encrypt traffic in a self-managed cluster
+
+If you're running a self-managed cluster, configure Transport Layer Security
+(TLS) to encrypt traffic between {agent}s, {fleet-server}, and other components
+in the {stack}.
+
+TIP: Our {ess-product}[hosted {ess}] on {ecloud} provides secure, encrypted
+connections out of the box!
+
+[discrete]
+[[prereqs]]
+== Prerequisites
+
+Configure security and generate certificates for the {stack}:
+
+. {ref}/security-minimal-setup.html[Set up minimal security for {es}]
+. {ref}/security-basic-setup.html[Set up basic security for the {stack}]
+. {ref}/security-basic-setup-https.html[Set up basic security for the {stack} plus secured HTTPS traffic]
+
+[discrete]
+[[generate-fleet-server-certs]]
+== Generate a custom certificate and private key for {fleet-server}
+
+This section describes how to use the `certutil` tool provided by {es}, but you
+can use whatever process you typically use to generate certificates.
+
+. (Optional) Generate a certificate authority (CA). Skip this step if you want
+to use an existing CA.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil ca --pem
+----
++
+This command creates a zip file that contains the CA certificate and key you'll
+use to sign the {fleet-server} certificate.
+
+. Use the certificate authority to generate certificates for {fleet-server}.
+For example:
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert \
+  --name fleet-server \
+  --ca-cert /path/to/ca/ca.crt \
+  --ca-key /path/to/ca/ca.key \
+  --dns your.host.name.here \
+  --ip 192.0.2.1 \
+  --pem
+----
++
+This command creates a certificate bundle that includes a `.crt` and `.key`
+file. You'll need these files later to encrypt traffic between {agent}s
+and {fleet-server}.
+
+. Store the files in a secure location until you are ready to use them.
+
+//QUESTION: Do we need to say anything here about key passwords not being supported?
+
+[discrete]
+== Encrypt traffic between {agent}s, {fleet-server}, and {es}
+
+{fleet-server} needs a CA certificate to connect securely to {es}. It also
+needs to expose a {fleet-server} certificate so other {agent}s can connect to it
+securely.
+
+. On the system running the {fleet-server} agent, add:
```

Comment:
Is this and the next step useful? I think _how_ the certificates are provided is gonna vary a lot, so I'm not sure this section is necessary or useful.

Reply:
Running through the steps again, I think it might be more useful for customers if I show filenames to set up the examples.